### PR TITLE
add value expander

### PR DESCRIPTION
**Pull request by vetinari**
This PR lets you transform value with a mapping function. 

It let's you e.g. read a .ini file and expand all ${NAME} in a value with the names value from the environment via `os.ExpandEnv` as `ValueMapper`.

### CODE REVIEW

File: ini.go [hunk 1]
```diff
@@ -127,6 +127,8 @@ type File struct {
 	looseMode bool
 
 	NameMapper
+
```
Comment (Unknwon) [inline ini.go]: Please remove this empty line.

File: README.md [hunk 2]
```diff
@@ -550,6 +550,26 @@ func main() {
...
+func main() {
+	cfg, err := ini.Load([]byte("[env]\nfoo = ${MY_VAR}\n")
+	cfg.ValueMapper = os.ExpandEnv
+	// ...
+	env := &Env{}
+	err = cfg.Section("env").MapTo(env)
+}
+```
+
+which would set the value of `env.Foo` to the value of the environment variable `MY_VAR`
```
Comment (Unknwon) [inline README.md]: Missing `.` at the end of line.

**Comment by Unknwon** [comment]
Thanks your PR!

Make some comments for few trivial changes.

**Comment by vetinari** [comment]
any chances this is merged?

**Pull request merged**